### ISSUE TITLE
FATAL: Error, expected inline copy of 'yaml' ... ModuleNotFoundError("No module named 'yaml'").

### ISSUE THREAD

**Issue by SimonNejmann** [Nuitka/Nuitka#1450]
I am installing the newest version of Nuitka on raspberry pi, but it complains about not being able to find "yaml". There is however a "yaml_35" directory where it looks for "yaml", and if I make a symbolic link then it works.

Steps to reproduce:

python3 -m venv env
source env/bin/activate
python3 -m pip install nuitka
python3 -m pip list
```
Package       Version
------------- -------
Nuitka        0.7.2
pip           20.3.4
pkg-resources 0.0.0
setuptools    44.1.1
```

cat hello-world.py
`print("Hello World")
`

python3 -m nuitka --standalone hello-world.py
```
Nuitka-Options:INFO: Used command line options: --standalone hello-world.py
FATAL: Error, expected inline copy of 'yaml' to be in '/home/pi/nuitka-test/env/lib/python3.9/site-packages/nuitka/build/inline_copy/yaml', error was: ModuleNotFoundError("No module named 'yaml'").
```

ls -l env/lib/python3.9/site-packages/nuitka/build/inline_copy/
```
total 36
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 appdirs
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 bin
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 glob2
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 jinja2_35
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 lib
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 markupsafe
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 tqdm
drwxr-xr-x 3 pi pi 4096 Feb 28 09:04 yaml_35
drwxr-xr-x 4 pi pi 4096 Feb 28 09:04 zstd
```

ln -s /home/pi/nuitka-test/env/lib/python3.9/site-packages/nuitka/build/inline_copy/yaml_35/ /home/pi/nuitka-test/env/lib/python3.9/site-packages/nuitka/build/inline_copy/yaml

python3 -m nuitka --standalone hello-world.py
```
Nuitka-Options:INFO: Used command line options: --standalone hello-world.py
Nuitka:INFO: Starting Python compilation with Nuitka '0.7.2' on Python '3.9' commercial None.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'pathlib' added 'ntpath,posixpath'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'importlib.metadata' added 'email'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'ctypes' added '_ctypes'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'locale' added '_bootlocale'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'encodings' added 'encodings.idna,encodings.base64_codec,encodings.hex_codec,encodings.rot_13'.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: gcc (gcc).
Nuitka-Scons:INFO: Backend linking program (no progress information available).
Nuitka-Scons:INFO: Compiled 10 C files using ccache.
Nuitka-Scons:INFO: Cached C files (using ccache) with result 'cache hit': 10
Nuitka:INFO: Keeping build directory 'hello-world.build'.
Nuitka:INFO: Successfully created 'hello-world.dist/hello-world'.
```

**Comment by Kereltje** [comment]
I ran into the same error on a fresh install of raspbian buster

python 3.7.3
pip 22.03
nuitka 0.7.2 installed with pip

I fixed the yaml error exactly like SimonNejman, but then Nuitka complained about a missing jinja2. Symlinking to jinja2_35 resulted in an error regarding missing ninja2.asyncsupport. 
Apparently Nuitka insists on using the module from inline_copy as pip installing jinja2 makes no difference. It works when a symlink to the jinja2 package installed with pip is created in inline_copy.

**Comment by kayhayen** [comment]
There were similar issues for the source distribution, which I believe I fixed by adding `pkg_resources`. What was happening there is that `jinja2` imports that, and this fails, and makes Jinja2 from inline copy unworkable.

Systems without pkg_resources are rare, but can be created and are actually desirable to have. I believe that 0.7.3 once it's out should work for you guys, as I am adding it to the setup.

**Comment by kayhayen** [comment]
Not sure how useful trying out factory is, but when installed via pip it should be better, maybe: https://nuitka.net/doc/factory.html

**Comment by kayhayen** [comment]
Ah, I didn't notice the hijacking of the issue, as such, the yaml inline copy as well as the Jinja2 ones, you have there one intended for Python3.5 and earlier. Using `virtualenv` I didn't manage to reproduce on my Linux, but maybe venv is different somehow, shall try that too.

**Comment by kayhayen** [comment]
I have a suspicion, that's pretty ugly actually:

Collecting nuitka
  Using cached Nuitka-0.7.2-py3-none-any.whl

The Nuitka wheel is not Python version specific beyond Python3, but the inline copies are in fact installed differently for Python 3.5 and 3.9, so you run into a false cache hit here, that is very likely what is going on. Very important finding.

**Comment by Kereltje** [comment]
I installed nuitka an a freshly installed raspbian buster. Surely there was no cached wheel on that machine?

**Comment by kayhayen** [comment]
It's the only explanation I have so far, and it's a valid bug. I fixed it on factory, can you try and update it from factory, where I pushed a fix for the issue I suspect. Should it persist, that would be strange. But naturally only 0.7.3 or current factory will be any better.

**Comment by SimonNejmann** [comment]
I just retested (in a new venv) with `python3 -m pip install -U --force-reinstall "https://github.com/Nuitka/Nuitka/archive/factory.zip"` instead.
So: python3 -m pip list
```
Package       Version
------------- -------
Nuitka        0.8rc3
pip           20.3.4
pkg-resources 0.0.0
setuptools    44.1.1
```
Using this version it compiled my test program without problems. Then for good measure I did:
python3 -m pip uninstall nuitka
python3 -m pip install nuitka
Got the 0.7.2 version back, and instantly had the problem again.

So, I don't know what you did, but it fixed the bug for me. :)

**Comment by Kereltje** [comment]
Yep, with me factory also works OK. Both on fresh install and after upgrading 0.7.2.
Many thanks!

**Comment by kayhayen** [comment]
Well, it's also going to be in the 0.7.3 hotfix later, these were actually two distinct issues, one package missing from inline copy always, and the cached wheels are traps. Similar symptoms do not mean the same underlying issue. But great, we got this figured out. I will keep you posted about when this will be on PyPI.

**Comment by kayhayen** [comment]
Both described changes are released with 0.7.3 hotifx.
